source/common/router/config_utility.cc

```diff
@@ -132,7 +132,7 @@ std::string ConfigUtility::parseDirectResponseBody(const envoy::config::route::v
     if (size < 0) {
       throw EnvoyException(absl::StrCat("cannot determine size of response body file ", filename));
     }
-    if (size > max_body_size_bytes) {
+    if (static_cast<unsigned int>(size) > max_body_size_bytes) {
```

Comment:
I would suggest:

if (static_cast<uint64_t>(size) > ...

File sizes may be larger than uint32_t max.  Casting to int can result in value truncation.

Reply:
Thanks for the suggestion @antoniovicente! pushed!